mscore/preferences.cpp

```diff
@@ -38,6 +38,38 @@ void Preferences::init(bool storeInMemoryOnly)
             _settings = new QSettings();
             }
 
+#if defined(Q_OS_WIN)
+      // Use the correct default GUI font on Windows. Qt 5.x and below use the deprecated function GetStockObject() with
+      // DEFAULT_GUI_FONT, which returns MS Shell Dlg 2 in 8 pt. MS Shell Dlg 2 is a virtual font that maps to Tahoma, which has
+      // not been the default Windows GUI font since 2006.
+      //
+      // The correct way to determine the default GUI font is to call SystemParametersInfoW() with SPI_GETNONCLIENTMETRICS and use
+      // the returned lfMessageFont structure. On all versions of Windows from Windows Vista through Windows 10, this typically
+      // returns Segoe UI in 9 pt.
+      //
+      // This problem is slated to be fixed in Qt 6, and the code here is equivalent to the code that's already been written for
+      // Qt 6 in QWindowsFontDatabase::systemDefaultFont(). For details, see: https://bugreports.qt.io/browse/QTBUG-58610
+      NONCLIENTMETRICSW ncm { sizeof(NONCLIENTMETRICSW) };
+      if (SystemParametersInfoW(SPI_GETNONCLIENTMETRICS, ncm.cbSize, &ncm, 0U)) {
+            const auto& font = ncm.lfMessageFont;
+            std::unique_ptr<std::remove_pointer<HDC>::type, void (*)(HDC)> dc(GetDC(NULL), [](HDC dc) { ReleaseDC(NULL, dc); });
+            const auto dpi = dc ? GetDeviceCaps(dc.get(), LOGPIXELSY) : 96;
+            const auto size = static_cast<int>(-font.lfHeight * 72 / dpi);
+            constexpr std::size_t numWeights = 9;
+            constexpr std::array<QFont::Weight, numWeights> weights { QFont::Thin, QFont::ExtraLight, QFont::Light, QFont::Normal, QFont::Medium, QFont::DemiBold, QFont::Bold, QFont::ExtraBold, QFont::Black };
+            const auto weight = weights[std::min<int>(std::max<int>(0, (font.lfWeight - 50) / 100), numWeights -1)];
+
+            QApplication::setFont(QFont(QString::fromWCharArray(font.lfFaceName), size, weight, font.lfItalic));
+            }
+
+      // If the current settings file is using the deprecated default GUI font, reset it so that the correct default font can be
+      // picked up automatically.
+      if (_settings->value(PREF_UI_THEME_FONTFAMILY) == "MS Shell Dlg 2") {
```

Comment:
Wouln't this also affect those that set it to this deliberatly?
Maybe also check for the default size (8) here?

Reply:
I considered that, but I deliberately left out the size check because (as I mentioned in the issue and in the commit message), the user shouldn't be setting the font to MS Shell Dlg 2 deliberately in the first place. If the user wants to use Tahoma (which is the actual font used), he or she should select Tahoma. MS Shell Dlg 2 was never meant to be a user-facing font, even before it was deprecated.

I think it's less confusing to the user to disallow MS Shell Dlg 2 in all cases rather than to allow it for all sizes except for 8.

Reply:
OK, fair enough